lib/widgets/message_list.dart

```diff
@@ -1268,15 +1275,27 @@ class MessageWithPossibleSender extends StatelessWidget {
           if (senderRow != null)
             Padding(padding: const EdgeInsets.fromLTRB(16, 2, 16, 0),
               child: senderRow),
-          EditStateMarker(
-            editState: message.editState,
+          Row(
+            crossAxisAlignment: CrossAxisAlignment.baseline,
+            textBaseline: localizedTextBaseline(context),
             children: [
-              Expanded(child: Column(
-                crossAxisAlignment: CrossAxisAlignment.stretch,
+              const SizedBox(width: 16),
+              Expanded(child: Stack(
                 children: [
-                  MessageContent(message: message, content: item.content),
-                  if ((message.reactions?.total ?? 0) > 0)
-                    ReactionChipsList(messageId: message.id, reactions: message.reactions!)
+                  Column(
+                    crossAxisAlignment: CrossAxisAlignment.stretch,
+                    children: [
+                      MessageContent(message: message, content: item.content),
+                      if ((message.reactions?.total ?? 0) > 0)
+                        ReactionChipsList(
+                          messageId: message.id,
+                          reactions: message.reactions!,
+                          // This copy of `editStateLabel` is invisible.
+                          // See docs for more details.
+                          trailingPlaceholder: editStateLabel),
+                    ]),
+                  if (editStateLabel != null)
+                    PositionedDirectional(bottom: 0, end: 0, child: editStateLabel),
```

Comment:
What happens if there are no reactions, but the message's own text ends with a full line? I think in this version that will end up having an overlap of the "EDITED" marker and the message text.

Reply:
Hmm, yeah, that could totally happen. It does look like that furthering this implementation to fix that as well can make it more complicated, unless we get a different solution.